### ISSUE TITLE
Trying to create new controller in monologue engine

### ISSUE THREAD

**Issue by codeundercoverdev** [jipiboily/monologue#274]
I'm trying to create a description page for blog post authors.  I've created a route:

      mount Monologue::Engine, at: '/blog'

      #create routes for monologue
      Monologue::Engine.routes.draw do
          get 'p/:name', to: 'authors#show', as: :author
      end

And I've created a controller in app/controllers/monologue:

    class Monologue::AuthorsController
	    def show
		    puts "in show method"
	    end
    end

I keep getting this error:

    undefined method `action' for Monologue::AuthorsController:Class

Long story short: how do I generate a controller within monologue?

**Comment by jipiboily** [comment]
It needs to inherit from a bad controller or ApplicationController



Jean-Philippe Boily
Human. Ruby developer.  | http://jipiboily.com/

**Comment by msevestre** [comment]
Probably  mean a " base " controller ☺

**Comment by codeundercoverdev** [comment]
I was able to fix it by doing this:

    module Monologue
    	class AuthorsController < Monologue::ApplicationController
	    	def show
		    	@rachel = "yes"
	    	end
	    end
    end

I had to nest the class in the Monologue module and had to have it inherit from the Monologue controller because there were conflicts with ApplicationController

**Comment by jipiboily** [comment]
@msevestre lol, yes! Autocorrect...I guess? :grimacing:

**Comment by jipiboily** [comment]
@codeundercoverdev glad you got it fixed!
